src/init.cpp

```diff
@@ -986,9 +986,6 @@ bool AppInitParameterInteraction()
     if (gArgs.GetArg("-prune", 0)) {
         if (gArgs.GetBoolArg("-txindex", DEFAULT_TXINDEX))
             return InitError(_("Prune mode is incompatible with -txindex."));
-        if (!g_enabled_filter_types.empty()) {
```

Comment:
This is problematic for a few reasons.

1) You can turn on the filter index and it will build in the background using the blocks on disk. This wouldn't work in pruned mode. It might be safe though to allow pruning if the block filter index tip is already within chain tip height - MIN_BLOCKS_TO_KEEP. But then we couldn't prune until the block filter index is in sync (it has its own sort of initial sync logic). Not prohibitively difficult though.
2) There still needs to be logic to fetch blocks from the network if the filters match, which I don't see here. Doing that in a blocking way would probably still be quite slow, especially with the default keypool size as large as it is.

Reply:
Hmm, right